lib/useful.ex

```diff
@@ -86,6 +86,35 @@ defmodule Useful do
     |> Enum.into(%{})
   end
 
+  @doc """
+  `get_in_default/3` Proxies `Kernel.get_in/2`
+  but allows setting a `default` value as the 3rd argument.
+
+  ## Examples
+
+      iex> map = %{name: "alex", data: %{age: 17, height: 185}}
+      iex> Useful.get_in_default(map, [:data, :age])
+      17
+      iex> Useful.get_in_default(map, [:data, :iq], 180)
+      180
+      iex> Useful.get_in_default(nil, [:unhappy, :path], "Happy!")
+      "Happy!"
+  """
+  def get_in_default(map, keys, default \\ nil) do
+    # Checking if it is a struct.
+    # If it is, we deeply convert it to map
+    map =
+      case Kernel.is_struct(map) do
+        true -> MapFromDeepStruct.from_deep_struct(map)
```

Comment:
Check https://github.com/dwyl/useful/pull/33#issuecomment-1361268904 to verify if `MapFroMDeepStruct` is applicable.